src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1845,6 +1845,10 @@ def prepare_inputs_for_generation(
         elif use_cache:
             cache_position = cache_position[-decoder_input_ids.shape[1] :]
 
+        # The `contiguous()` here is necessary to have a static stride during decoding. torchdynamo otherwise
+        # recompiles graphs as the stride of the inputs is a guard. Ref: https://github.com/huggingface/transformers/pull/29114
+        decoder_input_ids = decoder_input_ids.contiguous()
+
```

Comment:
In this case is contiguous enough? CF :
```python3
                # This `clone` call is needed to avoid recapturing cuda graphs with `torch.compile`'s  `mode="reduce-overhead`, as otherwise the input `position_ids` would have various stride during the decoding. Here, simply using `.contiguous()` is not sufficient as in the batch size = 1 case, `position_ids` is already contiguous but with varying stride which retriggers a capture.
                position_ids = position_ids.clone(memory_format=torch.contiguous_format)
```

Reply:
Here's tok/s for 4 different long-form audio clips, where run 1 includes the first compilation step:
```
Run 1: 3.8338281372199807
Run 2: 185.6919590403688
Run 3: 341.79757393227516
Run 4: 394.0163927342558
```
=> we get the expected speed-up from run 3 onwards